seqr/utils/vcf_utils.py

```diff
@@ -0,0 +1,70 @@
+import re
+
+from collections import defaultdict
+
+from seqr.utils.middleware import ErrorsWarningsException
+from seqr.utils.file_utils import file_iter
+
+BLOCK_SIZE = 65536
+
+EXPECTED_META_FIELDS ={
+    'INFO': {
+        'AC': 'Integer',
+        'AN': 'Integer',
+        'AF': 'Float'
+    },
+    'FORMAT': {
+        'AD': 'Integer',
+        'DP': 'Integer',
+        'GQ': 'Integer',
+        'GT': 'String'
+    }
+}
+
+REQUIRED_HEADERS = ['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT']
+
+
+def _validate_vcf_header(header):
+    missing_headers = [h for h in REQUIRED_HEADERS if h not in header ]
+    if missing_headers:
+        missing_fields = ', '.join(missing_headers)
+        raise ErrorsWarningsException([f'Missing required VCF header field(s) {missing_fields}.'], [])
+
+
+def _validate_vcf_meta(meta):
+    for field, sub_field_meta in EXPECTED_META_FIELDS.items():
+        for sub_field, field_type in sub_field_meta.items():
+            if meta.get(field, {}).get(sub_field) != field_type:
+                error = f'VCF header field {field}.{sub_field} and meta information Type={field_type} is expected.'
+                raise ErrorsWarningsException([error], [])
+
+
+def _get_vcf_meta_data(line):
+    r = re.search(r'##(?P<field>.*?)=<ID=(?P<id>[^,]*).*Type=(?P<type>[^,]*).*>$', line)
+    if r:
+        matches = r.groupdict()
+        return matches['field'], {matches['id']: matches['type']}
```

Comment:
splitting this into a tuple isn't so helpful, I think it would be neater to just have it return the dict and then do the formatting bellow with `meta[meta_row['field']].update({meta_row['id']: meta_row['type']})`

Reply:
Okay